File: Lib/DotNet/aaaSoft.Helpers.SnmpHelper/aaaSoft.Helpers.SnmpHelper/Model/hrStorageEntry.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace aaaSoft.Helpers.Model
{
    public class hrStorageEntry
    {
        public enum hrStorageTypes
        {
            hrStorageOther = 1,
            hrStorageRam = 2,
            hrStorageVirtualMemory = 3,
            hrStorageFixedDisk = 4,
            hrStorageRemovableDisk = 5,
            hrStorageFloppyDisk = 6,
            hrStorageCompactDisc = 7,
            hrStorageRamDisk = 8,
            hrStorageFlashMemory = 9,
            hrStorageNetworkDisk = 10
        }
        /// <summary>
        /// 驱动器序号
        /// </summary>
        public Int32 hrStorageIndex;
        /// <summary>
        /// 驱动器类型
        /// </summary>
        public hrStorageTypes hrStorageType;
        /// <summary>
        /// 驱动器描述
        /// </summary>
        public String hrStorageDescr;
        /// <summary>
        /// 驱动器分配单元大小
        /// </summary>
        public Int32 hrStorageAllocationUnits;
        /// <summary>
        /// 驱动器大小(单位：分配单元数量)
        /// </summary>
        public Int32 hrStorageSize;
        /// <summary>
        /// 驱动器已使用大小(单位：分配单元数量)
        /// </summary>
        public Int32 hrStorageUsed;
        /// <summary>
        /// 获取驱动器大小(单位：字节)
        /// </summary>
        public Int64 StorageSize { get { return 1L * hrStorageSize * hrStorageAllocationUnits; } }
        /// <summary>
        /// 获取驱动器已使用大小(单位：字节)
        /// </summary>
        public Int64 StorageUsed { get { return 1L * hrStorageUsed * hrStorageAllocationUnits; } }
        /// <summary>
        /// 获取驱动器使用率
        /// </summary>
        public Int32 StorageUsage { get { return hrStorageUsed * 100 / hrStorageSize; } }

        /// <summary>
        /// 从dataRow得到StorageEntry对象
        /// </summary>
        /// <param name="dataRow"></param>
        /// <returns></returns>
        public static hrStorageEntry FromDataRow(System.Data.DataRow dataRow)
        {
            var hrStorageIndex = Convert.ToInt32(dataRow["hrStorageIndex"]);
            var hrStorageType = (hrStorageTypes)Enum.Parse(typeof(hrStorageTypes), dataRow["hrStorageType"].ToString());
            var hrStorageDescr = aaaSoft.Helpers.StringHelper.GetLeftString(Convert.ToString(dataRow["hrStorageDescr"]), " ").Replace("\0", "");
            var hrStorageAllocationUnits = Convert.ToInt32(dataRow["hrStorageAllocationUnits"]);
            var hrStorageSize = Convert.ToInt32(dataRow["hrStorageSize"]);
            var hrStorageUsed = Convert.ToInt32(dataRow["hrStorageUsed"]);

            var se = new hrStorageEntry()
            {
                hrStorageIndex = hrStorageIndex,
                hrStorageType = hrStorageType,
                hrStorageDescr = hrStorageDescr,
                hrStorageAllocationUnits = hrStorageAllocationUnits,
                hrStorageSize = hrStorageSize,
                hrStorageUsed = hrStorageUsed,
            };
            return se;
        }
    }
}

File: trunk/Lib/Java/com.scbeta.helpers.SnmpHelper/src/com/scbeta/helpers/Model/hrStorageEntry.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package com.scbeta.helpers.Model;

import com.scbeta.helpers.ConsoleHelper;
import com.scbeta.helpers.MapDataTableHelper;
import com.scbeta.helpers.StringHelper;
import java.util.AbstractMap.SimpleEntry;
import java.util.LinkedList;
import java.util.List;
import java.util.Map;

/**
 *
 * @author aaa
 */
public class hrStorageEntry {

    public enum hrStorageTypes {

        hrStorageOther("1.3.6.1.2.1.25.2.1.1"), hrStorageRam("1.3.6.1.2.1.25.2.1.2"), hrStorageVirtualMemory("1.3.6.1.2.1.25.2.1.3"), hrStorageFixedDisk("1.3.6.1.2.1.25.2.1.4"), hrStorageRemovableDisk("1.3.6.1.2.1.25.2.1.5"), hrStorageFloppyDisk("1.3.6.1.2.1.25.2.1.6"), hrStorageCompactDisc("1.3.6.1.2.1.25.2.1.7"), hrStorageRamDisk("1.3.6.1.2.1.25.2.1.8"), hrStorageFlashMemory("1.3.6.1.2.1.25.2.1.9"), hrStorageNetworkDisk("1.3.6.1.2.1.25.2.1.10");
        private String hrStorageTypeString;

        private hrStorageTypes(String hrStorageTypeString) {
            this.hrStorageTypeString = hrStorageTypeString;
        }

        public static hrStorageTypes fromValue(String value) {
            for (hrStorageTypes tmp : hrStorageTypes.values()) {
                if (tmp.toValue().equals(value)) {
                    return tmp;
                }
            }
            return null;
        }

        public String toValue() {
            return hrStorageTypeString;
        }
    }
    // 驱动器序号
    public int hrStorageIndex;
    // 驱动器类型
    public hrStorageTypes hrStorageType;
    // 驱动器描述
    public String hrStorageDescr;
    // 驱动器分配单元大小
    public int hrStorageAllocationUnits;
    // 驱动器大小(单位：分配单元数量)
    public int hrStorageSize;
    // 驱动器已使用大小(单位：分配单元数量)
    public int hrStorageUsed;

    // 获取驱动器大小(单位：字节)
    public long getStorageSize() {
        return 1L * hrStorageSize * hrStorageAllocationUnits;
    }
    // 获取驱动器已使用大小(单位：字节)

    public long getStorageUsed() {
        return 1L * hrStorageUsed * hrStorageAllocationUnits;
    }
    // 获取驱动器使用率

    public int getStorageUsage() {
        return hrStorageUsed * 100 / hrStorageSize;
    }

    // 从Map对象得到实体数组
    public static hrStorageEntry[] fromMap(Map<String, List<String>> map) {
        int rowCount = MapDataTableHelper.GetRowCount(map);

        List<hrStorageEntry> EntryList = new LinkedList<hrStorageEntry>();
        for (int i = 0; i <= rowCount - 1; i++) {
            try {
                int hrStorageIndex = Integer.parseInt(map.get("hrStorageIndex").get(i));
                hrStorageTypes hrStorageType = hrStorageTypes.fromValue(map.get("hrStorageType").get(i));
                String hrStorageDescr = StringHelper.GetStringFromHexString(map.get("hrStorageDescr").get(i));
                hrStorageDescr = StringHelper.GetLeftString(hrStorageDescr," ");
                int hrStorageAllocationUnits = Integer.parseInt(map.get("hrStorageAllocationUnits").get(i));
                int hrStorageSize = Integer.parseInt(map.get("hrStorageSize").get(i));
                int hrStorageUsed = Integer.parseInt(map.get("hrStorageUsed").get(i));

                hrStorageEntry se = new hrStorageEntry();
                se.hrStorageIndex = hrStorageIndex;
                se.hrStorageType = hrStorageType;
                se.hrStorageDescr = hrStorageDescr;
                se.hrStorageAllocationUnits = hrStorageAllocationUnits;
                se.hrStorageSize = hrStorageSize;
                se.hrStorageUsed = hrStorageUsed;
                EntryList.add(se);
            } catch (Exception ex) {
            }
        }
        return EntryList.toArray(new hrStorageEntry[0]);
    }
}
